Kernel/Output/HTML/Layout.pm

```diff
@@ -2919,6 +2920,50 @@ sub JSONReply {
     );
 }
 
+=head2 AJAXException()
+
+=for stopwords customizable
+
+Throws an exception with customizable status code and body. Status can either be a client error (400-499) or a server error (500-599), see L<HTTP::Status> for details.
+
+    $LayoutObject->AJAXException(
+        StatusCode => 500,           # optional, default: 500
+        Body       => 'Some text',   # optional, data to pass as response body
+    );
+
+=cut
+
+sub AJAXException {
+    my ( $Self, %Param ) = @_;
+
+    # if given, check validity of status code
+    # allowed are client errors (400-499) and server errors (500-599)
+    my $StatusCode;
+    if ( $Param{StatusCode} ) {
+        if ( !is_client_error( $Param{StatusCode} ) && !is_server_error( $Param{StatusCode} ) ) {
+            $Kernel::OM->Get('Kernel::System::Log')->Log(
+                Priority => 'error',
+                Message  => "Status code $Param{StatusCode} is not a valid HTTP error status!",
+            );
+            return;
+        }
+        $StatusCode = $Param{StatusCode};
+    }
+    $StatusCode ||= 500;
```

Comment:
Being a bit nitpickey here, I suggest to write the assignment as
`my $StatusCode = $Param{StatusCode} || 500;`
The reasoning is that variables should have limited scope.

Reply:
But how do you then access the variable for usage in `Plack::Response->new()`, if you scope it to the if-Block?

Reply:
What do you think about the updated version I just pushed?

Reply:
Excellent, that is better than mine. First handle the default case and then the sanity check.